File: java2csharp/java/Samsung/ProfessionalAudio.FloatingConotrller.WidgetsLibrary/refactor/FcSapaModel.java
package com.samsung.android.sdk.professionalaudio.widgets.refactor;

import com.samsung.android.sdk.professionalaudio.app.SapaAppInfo;

import java.util.List;

/**
 * @brief Interface for model to communicate with Sapa API
 */
public interface FcSapaModel {

    /**
     * @brief Register new application in the model
     *
     * The application might be either main or ordinary, depending on the <code>mode</code>
     * parameter.
     *
     * @param appInfo
     * @param mode
     */
    void insertApp(SapaAppInfo appInfo, int mode);

    /**
     * @brief Unregister application from the model
     *
     * @param instanceId
     * @param mode
     */
    void removeApp(String instanceId, int mode);

    /**
     * @brief Update application from the model
     *
     * @param appInfo
     * @param mode
     */
    void updateApp(SapaAppInfo appInfo, int mode);

    /**
     * @brief Clear the model to the empty state
     *
     * It might be called when main application is shut down
     */
    void clear();

    /**
     * Order applications to display them properly in FloatingController
     *
     * @param trackOrder applications order according to their position on tracks
     */
    void orderApps(List<String> trackOrder);

    /**
     * @brief Set given application as the active one
     *
     * Active application is the one that displays floating controller
     *
     * @param appInfo
     */
    void setActiveApp(SapaAppInfo appInfo);

    void setFreezeApps(List<String> freezeList);
}

File: java2csharp/csharp/Samsung/ProfessionalAudio.FloatingConotrller.WidgetsLibrary/refactor/FcSapaModel.cs
using System.Collections.Generic;

namespace com.samsung.android.sdk.professionalaudio.widgets.refactor
{

	using SapaAppInfo = com.samsung.android.sdk.professionalaudio.app.SapaAppInfo;

	/// <summary>
	/// @brief Interface for model to communicate with Sapa API
	/// </summary>
	public interface FcSapaModel
	{

		/// <summary>
		/// @brief Register new application in the model
		/// 
		/// The application might be either main or ordinary, depending on the <code>mode</code>
		/// parameter.
		/// </summary>
		/// <param name="appInfo"> </param>
		/// <param name="mode"> </param>
		void insertApp(SapaAppInfo appInfo, int mode);

		/// <summary>
		/// @brief Unregister application from the model
		/// </summary>
		/// <param name="instanceId"> </param>
		/// <param name="mode"> </param>
		void removeApp(string instanceId, int mode);

		/// <summary>
		/// @brief Update application from the model
		/// </summary>
		/// <param name="appInfo"> </param>
		/// <param name="mode"> </param>
		void updateApp(SapaAppInfo appInfo, int mode);

		/// <summary>
		/// @brief Clear the model to the empty state
		/// 
		/// It might be called when main application is shut down
		/// </summary>
		void clear();

		/// <summary>
		/// Order applications to display them properly in FloatingController
		/// </summary>
		/// <param name="trackOrder"> applications order according to their position on tracks </param>
		void orderApps(IList<string> trackOrder);

		/// <summary>
		/// @brief Set given application as the active one
		/// 
		/// Active application is the one that displays floating controller
		/// </summary>
		/// <param name="appInfo"> </param>
		SapaAppInfo ActiveApp {set;}

		IList<string> FreezeApps {set;}
	}

}